azurelinuxagent/ga/cgroupcontroller.py

```diff
@@ -0,0 +1,175 @@
+        return self._counter
+
+    @property
+    def instance(self):
+        return self._instance
+
+    @property
+    def value(self):
+        return self._value
+
+    @property
+    def report_period(self):
+        return self._report_period
+
+
+class MetricsCategory(object):
+    MEMORY_CATEGORY = "Memory"
+    CPU_CATEGORY = "CPU"
+
+
+class MetricsCounter(object):
+    PROCESSOR_PERCENT_TIME = "% Processor Time"
+    THROTTLED_TIME = "Throttled Time"
+    TOTAL_MEM_USAGE = "Total Memory Usage"
+    ANON_MEM_USAGE = "Anon Memory Usage"
+    CACHE_MEM_USAGE = "Cache Memory Usage"
+    MAX_MEM_USAGE = "Max Memory Usage"
+    SWAP_MEM_USAGE = "Swap Memory Usage"
+    MEM_THROTTLED = "Total Memory Throttled Events"
+    AVAILABLE_MEM = "Available MBytes"
+    USED_MEM = "Used MBytes"
+
+
+class _CgroupController(object):
+    def __init__(self, name, cgroup_path):
+        """
+        Initialize _data collection for the controller
+        :param: name: Name of the CGroup
+        :param: cgroup_path: Path of the controller
+        :return:
+        """
+        self.name = name
+        self.path = cgroup_path
+
+    def __str__(self):
+        return "{0} [{1}]".format(self.name, self.path)
+
+    def _get_cgroup_file(self, file_name):
+        return os.path.join(self.path, file_name)
+
+    def _get_file_contents(self, file_name):
+        """
+        Retrieve the contents of file.
+
+        :param str file_name: Name of file within that metric controller
+        :return: Entire contents of the file
+        :rtype: str
+        """
+        parameter_file = self._get_cgroup_file(file_name)
+
+        return fileutil.read_file(parameter_file)
+
+    def _get_parameters(self, parameter_name, first_line_only=False):
+        """
+        Retrieve the values of a parameter from a controller.
+        Returns a list of values in the file.
+
+        :param first_line_only: return only the first line.
+        :param str parameter_name: Name of file within that metric controller
+        :return: The first line of the file, without line terminator
+        :rtype: [str]
+        """
+        result = []
+        try:
+            values = self._get_file_contents(parameter_name).splitlines()
+            result = values[0] if first_line_only else values
+        except IndexError:
+            parameter_filename = self._get_cgroup_file(parameter_name)
+            logger.error("File {0} is empty but should not be".format(parameter_filename))
+            raise CGroupsException("File {0} is empty but should not be".format(parameter_filename))
+        except Exception as e:
+            if isinstance(e, (IOError, OSError)) and e.errno == errno.ENOENT:  # pylint: disable=E1101
+                raise e
+            parameter_filename = self._get_cgroup_file(parameter_name)
+            raise CGroupsException("Exception while attempting to read {0}".format(parameter_filename), e)
+        return result
+
+    def is_active(self):
```

Comment:
I updated this method to check 'cgroup.procs' instead of 'tasks' 
There is no 'tasks' file in cgroup v2, so I switched this to cgroup.procs so that v1 and v2 had the same approach. 
More details on cgroup.procs in v1/v2 - 
v1: https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v1/cgroups.html
v2: https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#core-interface-files